From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 18:52:29 -0400
Subject: [infiniband] compat header cleanups
Message-id: <20100510185013.2781.53466.sendpatchset@prarit.bos.redhat.com>
Patchwork-id: 24981
O-Subject: [RHEL5 PATCH 21/27] compat.h cleanup: infiniband driver changes
Bugzilla: 546740
RH-Acked-by: Doug Ledford <dledford@redhat.com>

>From 07a9dc1ce1a8bd37b80c2947794af1abb7f8eab1 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 09:48:12 -0400
Subject: [PATCH] infiniband changes

The changes in infiniband are NOT due to the elimination of compat files in
the infiniband driver.  They are a result of modifications made elsewhere
in the kernel.

Resolves 546740.

diff --git a/drivers/infiniband/hw/amso1100/c2.h b/drivers/infiniband/hw/amso1100/c2.h
index fa58200..020916f 100644
--- a/drivers/infiniband/hw/amso1100/c2.h
+++ b/drivers/infiniband/hw/amso1100/c2.h
@@ -405,25 +405,6 @@ struct c2_port {
  */
 #define C2_ADAPTER_PCI_REGS_OFFSET 0x10000
 
-#ifndef readq
-static inline u64 readq(const void __iomem * addr)
-{
-	u64 ret = readl(addr + 4);
-	ret <<= 32;
-	ret |= readl(addr);
-
-	return ret;
-}
-#endif
-
-#ifndef writeq
-static inline void __raw_writeq(u64 val, void __iomem * addr)
-{
-	__raw_writel((u32) (val), addr);
-	__raw_writel((u32) (val >> 32), (addr + 4));
-}
-#endif
-
 #define C2_SET_CUR_RX(c2dev, cur_rx) \
 	__raw_writel(cpu_to_be32(cur_rx), c2dev->mmio_txp_ring + 4092)
 
diff --git a/drivers/infiniband/include/linux/if_vlan.h b/drivers/infiniband/include/linux/if_vlan.h
index 0c68427..a4394c5 100644
--- a/drivers/infiniband/include/linux/if_vlan.h
+++ b/drivers/infiniband/include/linux/if_vlan.h
@@ -3,17 +3,6 @@
 
 #include_next <linux/if_vlan.h>
 
-static inline struct net_device *vlan_group_get_device(struct vlan_group *vg, int vlan_id)
-{
-	return vg->vlan_devices[vlan_id];
-}
-
-static inline void vlan_group_set_device(struct vlan_group *vg, int vlan_id,
-					 struct net_device *dev)
-{
-	vg->vlan_devices[vlan_id] = dev;
-}
-
 #define vlan_dev_info(x) VLAN_DEV_INFO(x)
 
 static inline u16 vlan_dev_vlan_id(const struct net_device *dev)
diff --git a/drivers/infiniband/include/linux/types.h b/drivers/infiniband/include/linux/types.h
deleted file mode 100644
index 855a346..0000000
--- a/drivers/infiniband/include/linux/types.h
+++ /dev/null
@@ -1,9 +0,0 @@
-#ifndef BACKPORT_LINUX_TYPES_TO_2_6_19
-#define BACKPORT_LINUX_TYPES_TO_2_6_19
-
-#include_next <linux/types.h>
-
-typedef __u16	__sum16;
-typedef __u32	__wsum;
-
-#endif
diff --git a/drivers/infiniband/include/linux/workqueue.h b/drivers/infiniband/include/linux/workqueue.h
index f1428c9..e0c67b0 100644
--- a/drivers/infiniband/include/linux/workqueue.h
+++ b/drivers/infiniband/include/linux/workqueue.h
@@ -3,10 +3,6 @@
 
 #include_next <linux/workqueue.h>
 
-struct delayed_work {
-	struct work_struct work;
-};
-
 static inline void
 backport_INIT_WORK(struct work_struct *work, void *func)
 {
@@ -46,6 +42,7 @@ int backport_schedule_delayed_work(struct delayed_work *work, unsigned long dela
 
 #undef INIT_WORK
 #define INIT_WORK(_work, _func) backport_INIT_WORK(_work, _func)
+#undef INIT_DELAYED_WORK
 #define INIT_DELAYED_WORK(_work, _func) INIT_WORK(&(_work)->work, _func)
 #define INIT_DELAYED_WORK_DEFERRABLE(_work, _func) INIT_DELAYED_WORK(_work, _func)
 
